datadog/resource_datadog_dashboard.go

```diff
@@ -4722,6 +4722,257 @@ func buildTerraformTimeseriesDefinition(datadogDefinition datadogV1.TimeseriesWi
 	return terraformDefinition
 }
 
+func getFormulaSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"formula_expression": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "String expression built from queries, formulas and functions.",
+				},
+				"limit": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					Description: "Options for limiting results returned.",
+					MaxItems:    1,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"count": {
+								Type:        schema.TypeInt,
+								Optional:    true,
+								Description: "Number of results to return",
+							},
+							"order": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Default:     "desc",
+								Description: "Direction of sort.",
+							},
+						},
+					},
+				},
+				"alias": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Expression alias.",
+				},
+			},
+		},
+	}
+}
+
+func getFormulaQuerySchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"metric_query": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					MaxItems:    1,
+					Description: "A timeseries formula and functions metrics query.",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"data_source": {
```

Comment:
AFAIU there is only one possible value here, should we omit it from the schema?

Reply:
There is only one possible value here for `metric_query`. I personally prefer to be consistent and require this field in all definitions (`event_query`, `process_query` and `metric_query`), vs. not having users declare this for `metric_query`.

I'll defer to your judgement if there is a "best practice" that I am missing if I don't remove it.

Reply:
Sounds goood!